### PR TITLE
Fix PG behaviour when specifying timestamps in the past

### PR DESCRIPTION
**Pull request by leplatrem**
I rewrote cleanly 4 four tests for storage backends:
- create with timestamp specified in future and past
- update with timestamp specified in future and past

I fixed the migrations of PR #604 

And I updated the migration of PR #649 to fix the behaviour of PG for collection timestamps when specified record timestamp were in the past.

### CODE REVIEW

**Comment by Natim** [comment]
I don't understand why you had to modify the 008_to_009 migration file. However it looks good to me.

**Comment by leplatrem** [comment]
> I don't understand why you had to modify the 008_to_009 migration file

In PR #604, we updated the SQL schema, but didn't bump its version. As a consequence, the migration 008-009 was never executed.

In PR #649, I replaced the orphan migration file 008-009 with a new one about collection timestamps (my fault).

This PR restores the migration of #604 on 008-009 and puts the one of #649 into a new one 009-010.

**Comment by Natim** [comment]
Ok makes sense :+1:

File: cliquet_docs/api/resource.rst [hunk 1]
```diff
@@ -424,6 +424,18 @@ A ``details`` attribute in the response provides the offending record and
...
+Timestamp
+---------
+
+When a record is created, the timestamp of the collection is incremented.
+
+It is possible to force the timestamp if the specified record has a
+``last_modified`` attribute.
+
+The specified value will be ignored if it is less than the current collection
+timestamp.
```
Comment (leplatrem) [inline cliquet_docs/api/resource.rst]: Incorrect statement.

File: cliquet_docs/api/resource.rst [hunk 2]
```diff
@@ -567,17 +583,23 @@ The consumer might decide to ignore it.
...
+Timestamp
+---------
+
+When a record is deleted, the timestamp of the collection is incremented.
+
+It is possible to force the timestamp by passing it in the
+querystring with ``?last_modified=<value>``.
+
+The specified value will be ignored if it is less than the current collection
+timestamp.
```
Comment (leplatrem) [inline cliquet_docs/api/resource.rst]: Incorrect statement.

File: cliquet_docs/api/resource.rst [hunk 3]
```diff
@@ -651,6 +674,20 @@ changed meanwhile, a ``412 Precondition failed`` error is returned.
...
+---------
+
+When a record is created or replaced, the timestamp of the collection is incremented.
+
+It is possible to force the timestamp if the specified record has a
+``last_modified`` attribute.
+
+The specified value will be ignored if:
+
+* it is less than the current collection timestamp;
```
Comment (leplatrem) [inline cliquet_docs/api/resource.rst]: Incorrect statement.

File: cliquet_docs/api/resource.rst [hunk 4]
```diff
@@ -758,6 +799,21 @@ If changing a record field violates a field unicity constraint, a
...
+
+When a record is modified, the timestamp of the collection is incremented.
+
+It is possible to force the timestamp if the specified record has a
+``last_modified`` attribute.
+
+
+The specified value will be ignored if:
+
+* it is less than the current collection timestamp;
```
Comment (leplatrem) [inline cliquet_docs/api/resource.rst]: Incorrect statement.

File: cliquet_docs/api/timestamps.rst [hunk 5]
```diff
@@ -68,3 +68,20 @@ The client can then choose to:
...
+
+In order to replicate the timestamps when importing existing records,
+it is possible to force the last modified values.
+
+When a timestamp is specified, it should be strictly greater than the current
+collection timestamp.
+
+It will be ignored if:
+
+* it is less than the current collection timestamp;
```
Comment (leplatrem) [inline cliquet_docs/api/timestamps.rst]: Incorrect statement.

**Comment by Natim** [comment]
r+

**Pull request merged**